### PR TITLE
Add docm to detect_variants workflow

### PR DESCRIPTION
**Pull request by dufeiyu**
This PR is to add docm to detect_variant workflow.

### CODE REVIEW

File: docm/GATK_haplotype_caller.cwl [hunk 1]
```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env cwl-runner
+
+cwlVersion: v1.0
+class: CommandLineTool
+label: "HaplotypeCaller (GATK 3.6)"
+baseCommand: ["/usr/bin/java", "-jar", "/opt/GenomeAnalysisTK.jar", "-T", "HaplotypeCaller"]
```
Comment (jasonwalker80) [inline docm/GATK_haplotype_caller.cwl]: I would include the java heap size in the command-line, ex. `"-Xmx8g", "-jar"`

File: docm/GATK_haplotype_caller.cwl [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    ["-gt_mode", "GENOTYPE_GIVEN_ALLELES",
+    "-o", { valueFrom: $(runtime.outdir)/docm_out.vcf }]
+inputs:
+    reference:
+        type: File
+        inputBinding:
+            prefix: "-R"
+            position: 1
+        secondaryFiles: [.fai, ^.dict]
+    normal_bam:
```
Comment (jasonwalker80) [inline docm/GATK_haplotype_caller.cwl]: Do we want to genotype the normal or only the tumor to identify the presence of DoCM mutations? I was under the impression we only wanted to identify somatic events, in this case DoCM mutations unique to the tumor BAM/CRAM.

File: detect_variants/combine_docm.cwl [hunk 3]
```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env cwl-runner
+
+cwlVersion: v1.0
+class: CommandLineTool
+label: "CombineVariants (GATK 3.6)"
+baseCommand: ["/usr/bin/java", "-jar", "/opt/GenomeAnalysisTK.jar", "-T", "CombineVariants"]
```
Comment (jasonwalker80) [inline detect_variants/combine_docm.cwl]: Same comment as below, add the heap size.

File: detect_variants/combine_docm.cwl [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+arguments:
```
Comment (jasonwalker80) [inline detect_variants/combine_docm.cwl]: I would define at minimum a 4GB RAM resource requirement. On full exome data sets, this often failed without a minimum RAM requirement. It should be equal to or greater than the heap size above.

File: docm/GATK_haplotype_caller.cwl (hunk 2 above)
Comment (dufeiyu) [inline docm/GATK_haplotype_caller.cwl]: Dave is ok with this. We are looking for >5 reads and >1% VAF on tumor with normal as reference (not strict somatic).

File: docm/GATK_haplotype_caller.cwl (hunk 1 above)
Comment (dufeiyu) [inline docm/GATK_haplotype_caller.cwl]: I will make the change.

File: detect_variants/combine_docm.cwl (hunk 3 above)
Comment (dufeiyu) [inline detect_variants/combine_docm.cwl]: I will make the change.

File: detect_variants/combine_docm.cwl (hunk 4 above)
Comment (dufeiyu) [inline detect_variants/combine_docm.cwl]: I will add 8GB here.

File: docm/GATK_haplotype_caller.cwl (hunk 2 above)
Comment (jasonwalker80) [inline docm/GATK_haplotype_caller.cwl]: what do you mean "with normal as reference"? It seems unnecessary to run HaplotypeCaller on the normal if only the tumor genotypes are what we want. HaplotypeCaller doesn't have a somatic mode.

Comment (dufeiyu) [inline docm/GATK_haplotype_caller.cwl]: According to Dave, he also checks the var count and vaf of normal if the tumor gets >5 reads and >1% VAF.

Comment (jasonwalker80) [inline docm/GATK_haplotype_caller.cwl]: 👍

**Comment by dufeiyu** [comment]
combine_docm.cwl is based on combine_variants.cwl. So I also added ram requirement and heap size for combine_variants.cwl.

**Comment by dufeiyu** [comment]
There is a duplicated RAM requirement in select_variants.cwl. I removed it.

File: detect_variants/detect_variants.cwl [hunk 5]
```diff
@@ -102,10 +105,39 @@ steps:
             vcf: combine/combined_vcf
         out:
             [filtered_vcf]
+    bgzip:
+        run: bgzip.cwl
+        in:
+            file: filter/filtered_vcf
+        out:
+            [bgzipped_file]
+    index:
```
Comment (tmooney) [inline detect_variants/detect_variants.cwl]: There are already steps in `detect_variants.cwl` named `bgzip` and `index`.  They'll need to be unique or else references like `bgzip/bgzipped_file` become ambiguous.

Comment (dufeiyu) [inline detect_variants/detect_variants.cwl]: good point. make the change.

**Pull request merged**